js/layout-javascript/news-feed-code.js

```diff
@@ -1479,6 +1479,7 @@ DynamicList.prototype.renderLoopHTML = function (iterateeCb) {
         requestAnimationFrame(render);
       } else {
         _this.$container.find('.new-news-feed-list-container').removeClass('loading').addClass('ready');
+        ellipsizeTextBox('.news-feed-item-description', 3);
```

Comment:
Is this the same library as before or a different one?

Also, can we use the same `data-line-clamp` attribute as we did with the previous solution so that users can configure this via HTML template?

Reply:
@tonytlwu 
No, this is not the same library. I'll add a simple function that reduces the inner HTML of the element and add a dots at the end. 

> Also, can we use the same data-line-clamp attribute as we did with the previous solution so that users can configure this via HTML template?

Yes, we can use any CSS selector. I'll add it right away.

Reply:
@tonytlwu 
Added `data-line-clamp` to the template.

Reply:
I see. I've reviewed the library as well. I see a few problems with it.

Which was the library we previously used?

Reply:
@tonytlwu 
We were using [this](https://github.com/josephschmitt/Clamp.js/) library previously.

Reply:
@YaroslavOvdii What other library have you tried?

I found these two, and I wonder if they would be helpful.

https://github.com/nzambello/ellipsed
https://github.com/makandra/superclamp

Reply:
@tonytlwu If I may ask. Should we make one more feature with line clamp?
I probably wasn't looking into this issue from the perspective of improvement.

As far as I tested both libs you suggested, a better option for the further improvement would be a [superclamp](https://github.com/makandra/superclamp) lib.

And [this](https://github.com/nzambello/ellipsed) is similar to our current solution. 

But if we want just fix this issue. I would prefer to use a simpler solution which we could easier adjust to our needs. 

I agree that our solution may look raw but it can change easily to our needs and if you give us requirements of how the function should work we can make it great.

About the new feature, we could, for example, allow users to implement clamp-on text elements on any widget with an option.

Reply:
> About the new feature, we could, for example, allow users to implement clamp-on text elements on any widget with an option.

This is an interesting approach. I can see something like this being useful. In our usual fashion, however, we'd implement this to be specific to the LFD first and roll it out more globally if people feel the need to control line clamping across the app.

I would therefore suggest using [superclamp](https://github.com/makandra/superclamp) to implement this for the LFD, using `data-line-clamp` attributes on the containers to control the maximum number of lines to show.

Although a self-rolled solution might also work, based on my comments on your last round of implementation I'm not confident in the quality. I'd therefore be more confident in using a library that's been used more and for longer.

Reply:
@tonytlwu I'll use the superclamp then. I'll add it by my next commit.